### ISSUE TITLE
Improvement request

### ISSUE THREAD

**Issue by crssi** [sblask/webextension-skip-redirect#78]
Lets take an example from #76 

When you visit `www.dnsstuff.com` (so you are in a session with this domain) and there is a request like `http://www.dnsstuff.com/tools#dnsReport|type=domain&&value=www.google.com`, then there is no point to SkipRdirect, since the original request goes to the same domain as your session domain is.
It doesn't help privacy nor security IMO to do so, it only leads to false positives.

In upper example you land then at `www.google.com`, when you actually shouldn't.

Can you improve SkipRequest to do no action is case when target domain s the same as your session domain?

If I have wrote this in non-understandable way, please let me know for further clarification.

Cheers

**Comment by sblask** [comment]
SkipRedirect is not about privacy or security. It's about skipping stupid intermediate pages. If there wouldn't be any skipping in your example, where would be and how reliable would the distinction be?

**Comment by crssi** [comment]
The point is not in a privacy or security, but it is in a false positives and how to avoid those.
I am really sorry but, can you rephrase your question, since I don't understand it. :(

**Comment by sblask** [comment]
So what you are saying is that if I am on `example.com` and click on a link to `example.com?url=somethingelse.com` it should not skip, only if it's a link to  `anotherexample.com?url=somethingelse.com` . Why would that be a reliable heuristic?

**Comment by crssi** [comment]
Yes, exactly.
I am just thinking out loud how to avoid false positives that leads to breakages and I am not saying nether that this is a good/final solution or a reliable heuristic. Its debatable.
I am sure that most of us (no real grounds in stats to be honest) are using you fine WE to avoid tracking over 3rd parties, but in such cases there is nothing revealed to 3rd parties.
Don't get me wrong, please, my intentions are solely for improvements and not to force you into something you do not want or believe.
Cheers

**Comment by sblask** [comment]
Not convinced. If someone wants to implement it and add an option for it, I wouldn't mind adding it, but I won't work on it and will close this for now.

**Comment by crssi** [comment]
Doh. Lost quite a lot of time when convincing you for "Skip redirects for URLs with the same public domain" implementation and have no intentions to go over it again and again. ;)
In this case I am sure it will always lead to a breakage, because it has no sense for a page to "redirect" to the same domain as the page itself is, otherwise that this is needed parameter and desired redirect for application to work as intended. It will brake almost every whois or mxtoolbox like pages.
I would say that it is generally true and hope for your reconsideration.

Cheers

**Comment by sblask** [comment]
Counter examples: Google, Facebook, GMX - they all use the same domain to do whatever they're doing. As I said, if someone implements it as an optional thing, I can include it.

**Comment by crssi** [comment]
True. You are right. Sorry for your time.
Cheers
